### ISSUE TITLE
Anchored controls scale incorrectly on multi-monitor setup using HDPI PerMonitorV2

### ISSUE THREAD

**Issue by Rsge** [dotnet/winforms#10958]
### .NET version

.NET 8.0

### Did it work in .NET Framework?

Not tested/verified

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

On .NET 6 the issue isn't present this severely but present in a different way

### Issue description

When creating a WinForms app targeting .NET 8.0 with an auto-scaleable element anchored to at least two opposing sites on a form (e.g. a button or text box) and the project being set to HDPI mode `PerMonitorV2`, the element will wildly change it's size (up to being completely invisible) when dragged between monitors with different HDPI settings (e.g. one scaled to default 100 %, one to 150 %).

### Setup
![grafik](https://github.com/dotnet/winforms/assets/23245821/ffb95d4c-df7a-4af6-bdc0-2669e91219a6)
![grafik](https://github.com/dotnet/winforms/assets/23245821/eecf184f-db10-4d76-88aa-6e8b54f91c12)
![grafik](https://github.com/dotnet/winforms/assets/23245821/f6a08dc3-bd92-4683-82d8-980be5adeee8)

#### Loaded on 100 % monitor
![grafik](https://github.com/dotnet/winforms/assets/23245821/fd1687f9-328a-4aba-9bd4-0fd7305103c8)
#### Dragged to 150 % monitor first time
![grafik](https://github.com/dotnet/winforms/assets/23245821/40b43398-59d6-467d-9b7b-1b2e66680037)
#### Dragged back to 100 % monitor first time
![grafik](https://github.com/dotnet/winforms/assets/23245821/b88dd9d4-73ea-4d82-b85d-7507d59eae3f)
#### Dragged back to 150 % monitor second time
![grafik](https://github.com/dotnet/winforms/assets/23245821/ff492040-a123-46f6-92d9-37e4accd7af9)
#### Dragged back to 100 % monitor second time
![grafik](https://github.com/dotnet/winforms/assets/23245821/a1ed4e63-3fe1-44e4-b76e-d1c39bb9fdf8)
#### Dragged back to 150 % monitor third time
Same as second time.

#### Loaded on 100 % monitor
Same as in .NET 8.
#### Dragged to 150 % monitor first time
![grafik](https://github.com/dotnet/winforms/assets/23245821/cbcec132-9243-47e6-b368-e26b716cedaf)
#### Dragged back to 100 % monitor first time
![grafik](https://github.com/dotnet/winforms/assets/23245821/2f01b52f-7ed0-4910-812d-139b2968c7bd)
#### Dragged to 150 % monitor second time
Same as first time.

### PS
I think this issue is different from e.g. #8157 because that one is explicitly about elements in SplitContainers, while mine is about elements directly on forms.

### Steps to reproduce

1. Create a new WinForms project targeting .NET 8.0.
2. Add a button to the form.
3. Drag it bigger so it "anchors" to the edges.
4. Set the "Anchor" property to "Top, Bottom, Left, Right".
5. Start the program on a 100 % scaled monitor.
6. Drag it to a higher scaled (e.g. 150 %) monitor.
7. Drag it back to the 100 % monitor.

Example Project:
[Test.zip](https://github.com/dotnet/winforms/files/14419738/Test.zip)

**Comment by John-Qiao** [comment]
@elachlan yes, we can repro this issue with .NET 8.0 and latest .NET 9.0.100-preview.3.24127.9 SDK, please check below video.

Launch test app on 100% primary monitor and then move form to 175% secondary monitor.

https://github.com/dotnet/winforms/assets/45864985/e09c4b32-0391-4fa3-8063-35ea34291873

In above video result, I tried to add the runtimeconfig.template.json file with below codes in test project, that can improve most problems, but not solve all issues.
```
{
  "configProperties": {
    "System.Windows.Forms.AnchorLayoutV2": true
  }
}
```

Launched on 100% primary monitor
![100-addJsonFile](https://github.com/dotnet/winforms/assets/45864985/c0b80507-710a-41fb-a5e6-078d9c0a71a3)

Moved to 175% secondary monitor
![MoveTo175Secondary-addJsonFile](https://github.com/dotnet/winforms/assets/45864985/53f1c36b-bb66-44f1-901c-6e2ff1d1ce64)

**Comment by WolfgangHG** [comment]
Attached is another .NET8 sample where moving the form from screen 1 at 100% to screen 2 at 125% and back causes broken layout.
[DpiAwarenessHighDpiModePMV2.zip](https://github.com/user-attachments/files/16179320/DpiAwarenessHighDpiModePMV2.zip)

I tested it with `System.Windows.Forms.AnchorLayoutV2` = `true` and `false` in "runtimeconfig.template.json":

```
{
  "configProperties": {
    "System.Windows.Forms.AnchorLayoutV2": true
  }
}
```

Both modes show different behavior, but the result is broken for both of them. This blocks us from supporting DPI awareness in our application (since our first attempt to do so in 2016 :-( ). 


This is the result for AnchorLayoutV2 = true"  - start with the top left image, a bit below to the right is the result of moving the form to the 125% screen, then below to the left the form back at 100% and so on: 
![AnchorLayoutV2_true](https://github.com/dotnet/winforms/assets/138049/24371bb3-ebb4-4a31-a223-8e5dfa388af2)

And this is the result for "AnchorLayoutV2 = false" (old behavior).
![AnchorLayoutV2_false](https://github.com/dotnet/winforms/assets/138049/b9465f7a-2e6e-40e1-83cc-8821f7aaff32)


I also noted that the form has different sizes sometimes when moving to the 125% screen.


I don't like the fact that this issue has a milestone "Future" - as I wrote above this blocks us from activating dpi awareness. I want to avoid redesigning all forms in our app and replace Anchor layouts with Dock.

**Comment by Dave-Lowndes** [comment]
I've also encountered this issue. [Bug report here](https://developercommunity.visualstudio.com/t/WinForm-with-non-default-anchored-contro/10735502) closed in favour of this one.
The PerMonitorV2 & PerMonitor settings were similarly affected for me. I've resorted to using the SystemAware setting as the best compromise until this is fixed.

**Comment by Rsge** [comment]
One and a half years later this still has the „Future“ tag?
I’d be really interested in a short explanation as to why this doesn’t seem to be of high priority.
I don’t want to stress, I just don’t understand the reasoning behind this decision yet, given it’s a complete feature not working under, I’d say, rather common circumstances, and want to see the bigger picture.

**Comment by coderb** [comment]
I'm hitting an issue where RDP back and forth with 100% vs 125% scaling causes right anchored controls to shift their right position calculation repeatedly.

It feels like the location of the anchor is being cumulatively miscalculated each time (perhaps based on DPI confusion due to callback sequencing?)

I've pretty much had this issue from the get go with PerMonitorV2 scaling and have stuck to SystemAware to date because of it.

This is a pretty high priority blocker for me.

Can we get some love on this one please?
